daemon/logger/fluentd/fluentd.go

```diff
@@ -0,0 +1,99 @@
+package fluentd
+
+import (
+	"io"
+	"strconv"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/docker/docker/daemon/logger"
+	"github.com/fluent/fluent-logger-golang/fluent"
+)
+
+type Fluentd struct {
+	tag           string
+	containerID   string
+	containerName string
+	writer        *fluent.Fluent
+}
+
+const name = "fluentd"
+
+const defaultHostName = "localhost"
+const defaultPort = 24224
+const defaultTagPrefix = "docker"
+
+func init() {
+	if err := logger.RegisterLogDriver(name, New); err != nil {
+		logrus.Fatal(err)
+	}
+}
+
+func New(ctx logger.Context) (logger.Logger, error) {
+	config := ctx.Config
+	port := defaultPort
+	host := defaultHostName
+	if config["port"] != "" {
+		portp, errp := strconv.Atoi(config["port"])
+		if errp != nil {
+			return nil, errp
+		}
+		port = portp
+	}
+	if config["host"] != "" {
+		host = config["host"]
+	}
+	tag := config["tag"]
```

Comment:
It seems not good enough for me.
I think we should not prohibit users to configure tags like `docker.id.0123456789ab`, even if it looks bad configuration.

In these pattern, we can use special variable or placeholder syntax, ex: `$ID`,  `$(id)`, `${id}` or `#{id}`. With placeholders, users can configure their tags by specifying `docker.id.${id}`.